core/drivers/crypto/se050/adaptors/utils/info.c

```diff
@@ -0,0 +1,220 @@
+		uint8_t vLength_FIPS_mode;
+		uint8_t vFIPS_mode;
+		uint8_t vTag_pre_perso_state;
+		uint8_t vLength_pre_perso_state;
+		uint8_t vBit_mask_of_pre_perso_state;
+		uint8_t vTag_ROM_ID;
+		uint8_t vLength_ROM_ID;
+		uint8_t vROM_ID[0x08];
+		uint8_t vStatus_Word_SW_[0x02];
+	} rsp = { 0 };
+	uint8_t *p = (uint8_t *)&rsp;
+	uint32_t len = sizeof(struct msg_rsp);
+	uint16_t dummy = sizeof(struct msg_rsp);
+
+	ret = GP_Select(ctx, p, 0, p, &dummy);
+	if (ret != SM_OK) {
+		LOG_E("Could not select ISD.");
+		goto exit;
+	}
+
+	ret = smCom_TransceiveRaw(ctx, cmd, sizeof(cmd), p, &len);
+	if (ret != SM_OK || len != sizeof(rsp)) {
+		LOG_MAU8_E("Error reading JCOP ID", p, sizeof(rsp));
+		goto exit;
+	}
+
+	LOG_I("SE050 JCOP4 Information:");
+	LOG_I("%s = 0x%02X", "Tag value - proprietary data 0xFE",
+	      rsp.vTag_value_proprietary_data);
+	LOG_I("%s = 0x%02X", "Length of following data 0x45",
+	      rsp.vLength_of_following_data);
+	LOG_MAU8_I("Tag card identification data",
+		   rsp.vTag_card_identification_data,
+		   sizeof(rsp.vTag_card_identification_data));
+	LOG_I("%s = 0x%02X", "Length of card identification data",
+	      rsp.vLength_of_card_identification_data);
+	LOG_I("%s = 0x%02X", "Tag configuration ID (Must be 0x01)",
+	      rsp.vTag_configuration_ID);
+	LOG_I("%s = 0x%02X", "Length configuration ID 0x0C",
+	      rsp.vLength_configuration_ID);
+	LOG_MAU8_I("Configuration ID",
+		   rsp.vConfiguration_ID, sizeof(rsp.vConfiguration_ID));
+
+	LOG_MAU8_I("OEF ID", &rsp.vConfiguration_ID[2], 2);
+	LOG_I("%s = 0x%02X", "Tag patch ID (Must be 0x02)", rsp.vTag_patch_ID);
+	LOG_I("%s = 0x%02X", "Length patch ID 0x08", rsp.vLength_patch_ID);
+	LOG_MAU8_I("Patch ID", rsp.vPatch_ID, sizeof(rsp.vPatch_ID));
+	LOG_I("%s = 0x%02X", "Tag platform build ID1 (Must be 0x03)",
+	      rsp.vTag_platform_build_ID1);
+	LOG_I("%s = 0x%02X", "Length platform build ID 0x18",
+	      rsp.vLength_platform_build_ID);
+	LOG_MAU8_I("Platform build ID",
+		   rsp.vPlatform_build_ID, sizeof(rsp.vPlatform_build_ID));
+	memcpy(jcop_platform_id, rsp.vPlatform_build_ID, 16);
+
+	LOG_I("%s = %s", "JCOP Platform ID", jcop_platform_id);
+	LOG_I("%s = 0x%02X", "Tag FIPS mode (Must be 0x05)",
+	      rsp.vTag_FIPS_mode);
+	LOG_I("%s = 0x%02X", "Length FIPS mode 0x01", rsp.vLength_FIPS_mode);
+	LOG_I("%s = 0x%02X", "FIPS mode var", rsp.vFIPS_mode);
+	LOG_I("%s = 0x%02X", "Tag pre-perso state (Must be 0x07)",
+	      rsp.vTag_pre_perso_state);
+	LOG_I("%s = 0x%02X", "Length pre-perso state 0x01",
+	      rsp.vLength_pre_perso_state);
+	LOG_I("%s = 0x%02X", "Bit mask of pre-perso state var",
+	      rsp.vBit_mask_of_pre_perso_state);
+
+	LOG_I("%s = 0x%02X", "Tag ROM ID (Must be 0x08)", rsp.vTag_ROM_ID);
+	LOG_I("%s = 0x%02X", "Length ROM ID 0x08", rsp.vLength_ROM_ID);
+	LOG_MAU8_I("ROM ID", rsp.vROM_ID, sizeof(rsp.vROM_ID));
+	LOG_MAU8_I("Status Word (SW)", rsp.vStatus_Word_SW_,
+		   sizeof(rsp.vStatus_Word_SW_));
+
+	status = kStatus_SSS_Success;
+exit:
+	return status;
```

Comment:
Minor: simplify by replacing `goto exit;` with `return kStatus_SSS_Fail;` above and using here `return kStatus_SSS_Success;`.

This comment also applies to `applet_get_id()`.

Reply:
ok